---
# Documentation: https://wowchemy.com/docs/managing-content/

title: "Menggunakan Facebook Prophet untuk meramal harga beras"
subtitle: ""
summary: ""
authors: [admin] 
tags: [perdagangan, ekonomi]
categories: []
date: 2025-08-02T11:00:00+07:00
lastmod: 2025-08-02T11:00:00+07:00
featured: false
draft: false

# Featured image
# To use, add an image named `featured.jpg/png` to your page's folder.
# Focal points: Smart, Center, TopLeft, Top, TopRight, Left, Right, BottomLeft, Bottom, BottomRight.
image:
  caption: ""
  focal_point: ""
  preview_only: false

# Projects (optional).
#   Associate this post with one or more of your projects.
#   Simply enter your project's folder or file name without extension.
#   E.g. `projects = ["internal-project"]` references `content/project/deep-learning/index.md`.
#   Otherwise, set `projects = []`.
projects: []
---



In [29]:
# Bikin class dulu
# Optimisasi sudah dilakukan dengan tangan di atas, jadi class-nya ga perlu optimisasi lagi. tinggal kalkulasi biasa.
import pandas as pd
import plotly.express as px
from prophet import Prophet

df = pd.read_excel('beras.xlsx')
df['month'] = pd.Series(pd.date_range(start='2018-07-01', end='2025-05-10', freq='ME'))
dfl=df.melt(id_vars=['month'], var_name='jenis', value_name='harga')
fig = px.line(dfl, x="month", y="harga", title='harga beras bulanan (PIHPS)',color='jenis',
              color_discrete_sequence=px.colors.qualitative.G10,template='plotly_dark')
fig.write_html('wew.html', full_html=False, include_plotlyjs='cdn')

In [37]:
df = pd.read_excel('beras.xlsx',sheet_name='fit')
df['ds'] = pd.Series(pd.date_range(start='2018-07-01', end='2025-05-10', freq='ME'))
m=Prophet()
m.fit(df)

11:14:51 - cmdstanpy - INFO - Chain [1] start processing
11:14:51 - cmdstanpy - INFO - Chain [1] done processing
11:14:51 - cmdstanpy - ERROR - Chain [1] error: terminated by signal 3221225657 


RuntimeError: Error during optimization! Command 'C:\Users\PC\anaconda3\Lib\site-packages\prophet\stan_model\prophet_model.bin random seed=35018 data file=C:\Users\PC\AppData\Local\Temp\tmp3ys49ec9\ifav61yl.json init=C:\Users\PC\AppData\Local\Temp\tmp3ys49ec9\p40rftvg.json output file=C:\Users\PC\AppData\Local\Temp\tmp3ys49ec9\prophet_modelrdil9rl8\prophet_model-20250802111451.csv method=optimize algorithm=newton iter=10000' failed: 

Kita tes modelnya yuk. Pertama, run kode di atas, lalu run default parameters dengan cara di bawah ini.

Sesuai prediksi, harga A (pangan) lebih tinggi daripada harga B (sandang) karena ekonomi negara ini lebih produktif membuat sandang daripada pangan.

Tentu saja kita bisa coba pakai angka lain. Let's say kita punya negara lain dengan karakteristik jumlah tenaga kerja 400 juta orang dan relatif lebih produktif agrikulturnya daripada pakaian jadi-nya. Oh, kita juga bisa misalkan wage level di negara ini lebih tinggi 3x lipat daripada negara sebelah.

In [5]:
ekon(L=400,w=3,f=0.8,g=0.2).hasil()

Goods market: 
 harga A adalah 3.75 dan harga B adalah 15.00 
 Jumlah produksi (A,B) adalah (160.0, 40.0) 
Labor market: 
 gaji adalah 3.00 
 dari total 400.00 pekerja, sebanyak 200.00 bekerja di industri A dan 200.00 di industri B 
makro: 
 PDB nominal adalah 1,200.00 dan 3.00 per kapita  
 PDB riil adalah 200.00 dan 0.50 per kapita 
 level harga adalah 6.00 
 welfare di negara ini adalah 80.00 dan 0.20 per kapita 


Katakanlah negara pertama kita sebut Home (H) sementara negara ke-2 kita sebut Foreign (F). Ciri utama adalah negara F lebih besar (400 pekerja dibanding 200) dan lebih produktif di agrikultur ketimbang garmen. Kita langsung lihat bahwa negara H membuat A=25 dan B=75, lebih condong banyak bikin B. Sementara itu, negara F lebih berat ke A, membuat A sebanyak 160 sementara B sebanyak 40, atau perbedaan 4 kali lipat.

Kita juga bisa bereksperimen dengan karakteristik lain, misalnya negara F lebih suka garmen sehingga $\beta=0.7$ sementara $\alpha=0.3$, dan lain sebagainya.

Nah pertanyaan lebih menariknya adalah, apa yang terjadi kalau kedua negara ini melakukan trade? Ntar ya saya bikinin lagi kalo postingan ini rame wkwk.